### PR TITLE
CDAP-5561 CDAP-5436 CDAP-5332 Misc system metadata fixes

### PR DESCRIPTION
**Pull request by poornachandra**
Delete non-relevant system metadata on update, fix bug in metadata delete and rename system metadata keys

Build - http://builds.cask.co/browse/CDAP-DUT3902-7

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/system/AbstractSystemMetadataWriter.java [hunk 1]
```diff
@@ -24,21 +24,27 @@
...
-  private static final String SCHEMA_FIELD_PROPERTY_PREFIX = "schema";
-  private static final String PLUGIN_KEY_PREFIX = "plugin";
-  private static final String PLUGIN_VERSION_KEY_PREFIX = "pluginversion";
+  protected static final String SCHEMA_FIELD_PROPERTY_PREFIX = "schema";
+  protected static final String PLUGIN_KEY_PREFIX = "plugin";
+  protected static final String PLUGIN_VERSION_KEY_PREFIX = "plugin-version";
   protected static final String TTL_KEY = "ttl";
   protected static final String DESCRIPTION = "description";
-  protected static final String CREATE_TIME = "createtime";
+  protected static final String CREATE_TIME = "creation-time";
```
Comment (bdmogal) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/system/AbstractSystemMetadataWriter.java]: nit: call it ``CREATION_TIME`` if the value is ``creation-time``?

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/system/AbstractSystemMetadataWriterTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import com.google.inject.util.Modules;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+
+/**
+ *
```
Comment (bdmogal) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/system/AbstractSystemMetadataWriterTest.java]: empty javadoc

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/system/AbstractSystemMetadataWriterTest.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    // Now remove TTL, and add dsType
+    datasetSystemMetadataWriter =
+      new DatasetSystemMetadataWriter(store, dsInstance, DatasetProperties.EMPTY, null, "dsType", "description2");
+    datasetSystemMetadataWriter.write();
+
+    expected =
+      new MetadataRecord(dsInstance, MetadataScope.SYSTEM,
+                         ImmutableMap.of(AbstractSystemMetadataWriter.DESCRIPTION, "description2",
+                                         AbstractSystemMetadataWriter.CREATE_TIME, String.valueOf(123456L),
+                                         "type", "dsType"),
```
Comment (bdmogal) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/system/AbstractSystemMetadataWriterTest.java]: is there a constant for ``type``?

**Comment by bdmogal** [comment]
minor comments. Rest LGTM

**Comment by poornachandra** [comment]
@bdmogal Addressed comments, please take another look.

**Comment by bdmogal** [comment]
LGTM

**Comment by poornachandra** [comment]
Will squash and merge after build

File: cdap-client-tests/src/test/java/co/cask/cdap/client/MetadataHttpHandlerTestRun.java [hunk 4]
```diff
@@ -600,39 +600,43 @@ public void testSystemMetadataRetrieval() throws Exception {
 
     Map<String, String> streamSystemProperties = getProperties(streamId, MetadataScope.SYSTEM);
     // Verify create time exists, and is within the past hour
-    Assert.assertTrue("Stream create time does not exist", streamSystemProperties.containsKey("createtime"));
-    long createTime = Long.parseLong(streamSystemProperties.get("createtime"));
+    final String creationTime = "creation-time";
+    String description = "description";
+    String schema = "schema";
+    String ttl = "ttl";
+    Assert.assertTrue("Stream create time does not exist", streamSystemProperties.containsKey(creationTime));
```
Comment (bdmogal) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/MetadataHttpHandlerTestRun.java]: ``"Expected creation time to exist but it does not"``?

File: cdap-client-tests/src/test/java/co/cask/cdap/client/MetadataHttpHandlerTestRun.java [hunk 5]
```diff
@@ -600,39 +600,43 @@ public void testSystemMetadataRetrieval() throws Exception {
 
     Map<String, String> streamSystemProperties = getProperties(streamId, MetadataScope.SYSTEM);
     // Verify create time exists, and is within the past hour
-    Assert.assertTrue("Stream create time does not exist", streamSystemProperties.containsKey("createtime"));
-    long createTime = Long.parseLong(streamSystemProperties.get("createtime"));
+    final String creationTime = "creation-time";
```
Comment (bdmogal) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/MetadataHttpHandlerTestRun.java]: shouldn't these be constants somewhere already?

File: cdap-client-tests/src/test/java/co/cask/cdap/client/MetadataHttpHandlerTestRun.java [hunk 6]
```diff
@@ -600,39 +600,43 @@ public void testSystemMetadataRetrieval() throws Exception {
...
       Maps.filterEntries(streamSystemProperties, new Predicate<Map.Entry<String, String>>() {
         @Override
         public boolean apply(Map.Entry<String, String> input) {
-          return !"createtime".equals(input.getKey());
+          return !creationTime.equals(input.getKey());
         }
       }));
 
-    // Update stream properties and verify metadata got updated (except create time)
+    // Update stream properties and verify metadata got updated (except creation time)
```
Comment (bdmogal) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/MetadataHttpHandlerTestRun.java]: and description?

File: cdap-client-tests/src/test/java/co/cask/cdap/client/MetadataHttpHandlerTestRun.java (hunk 5 above)
Comment (poornachandra) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/MetadataHttpHandlerTestRun.java]: Today we don't have a use case to expose these as constants.

**Comment by bdmogal** [comment]
LGTM

**Comment by poornachandra** [comment]
Will squash and merge

**Comment by poornachandra** [comment]
Build is green, merging.

**Pull request merged**